### ISSUE TITLE
Exec.exec() fails in Electron app (ws.client not a constructor)

### ISSUE THREAD

**Issue by cabrinoob** [kubernetes-client/javascript#123]
Hi,
I'am using your lib in an Angular 6 application. I'd like to use the Exec.exec() method to display a terminal in my app to open a shell command in a pod.

My problem is that it seems to have an error with the websocket connect() method

When I call this from my Angular app : 

```
let Exec = new k8s.Exec(this.kubeConfig)
Exec.exec('default','myservice-6c5c9995b7-h79pq','myservice','bash',this.outStream,this.errStream,this.inStream,true)
```

I have this error :

```
Error: Uncaught (in promise): TypeError: ws.client is not a constructor
TypeError: ws.client is not a constructor
    at WebSocketHandler.connect (web-socket-handler.js:65)
    at Exec.<anonymous> (exec.js:30)
    at Generator.next (<anonymous>)
    at exec.js:7
    at new ZoneAwarePromise (zone-mix.js:891)
    at push../node_modules/@kubernetes/client-node/dist/exec.js.__awaiter (exec.js:3)
    at Exec.exec (exec.js:19)
```

It's working fine in a nodeJS app.

**Comment by flyboarder** [comment]
The library should work from an electron node.js instance.

**Comment by cabrinoob** [comment]
I was'nt precise enough in my previous post : My Angular app is wrapped in an Electron application and I have this error message.

**Comment by itowlson** [comment]
Can you provide a minimal project that reproduces the problem?  (E.g. a patch to an Electron baseline app.)

**Comment by cabrinoob** [comment]
@itowlson : ok i'll try to give you that asap
Thank you

**Comment by brendandburns** [comment]
@cabrinoob there is this PR here:

**Comment by itowlson** [comment]
@brendandburns That PR addressed an Electron issue shelling out to an auth provider.  The error @cabrinoob is seeing looks more like a type level error.  I'm wondering if there is a versioning issue in play, or something like that.

**Comment by edvald** [comment]
The issue is likely that the `websocket` library used has a native dependency. I came across this as well in a different context. I suggest switching to https://github.com/websockets/ws which has the binary dependencies as optional addons for performance improvement (which I do not think is very important for this library).

**Comment by gaogaoSpark** [comment]
I came across  the same problem .how to Solve this problem？

web-socket-handler.js?f1bc:80 Uncaught (in promise) TypeError: ws.client is not a constructor
    at WebSocketHandler.connect (web-socket-handler.js?f1bc:80)
    at Exec.eval (exec.js?0f84:28)
    at Generator.next (<anonymous>)
    at eval (tslib.es6.js?4d3a:71)
    at new Promise (<anonymous>)
    at Object.__awaiter (tslib.es6.js?4d3a:67)
    at Exec.exec (exec.js?0f84:17)
    at eval (main.js?1c90:26)
    at <anonymous>

**Comment by brendandburns** [comment]
Closing since I believe this was fixed by #132, please re-open if there are still issues.
